core/src/main/java/org/kiwix/kiwixmobile/core/error/ErrorActivity.kt

```diff
@@ -0,0 +1,236 @@
+          FileProvider.getUriForFile(
+            this@ErrorActivity,
+            applicationContext.packageName + ".fileprovider",
+            file
+          )
+        addFlags(android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION)
+        putExtra(android.content.Intent.EXTRA_STREAM, path)
+      }
+    }
+  }
+
+  private fun buildBody(): String {
+    return """ 
+    $initialBody
+      
+    ${if (allowCrash.isChecked && exception != null) exceptionDetails() else ""}
+    ${if (allowZims.isChecked) zimFiles() else ""}
+    ${if (allowLanguage.isChecked) languageLocale() else ""}
+    ${if (allowDeviceDetails.isChecked) deviceDetails() else ""}
+    ${if (allowFileSystemDetails.isChecked) systemDetails() else ""}
+    """.trimIndent()
+  }
+
+  private fun exceptionDetails(): String =
+    """
+    Exception Details:
+    ${toStackTraceString(exception!!)}
+    """.trimIndent()
+
+  private fun zimFiles(): String {
+    val allZimFiles = bookDao.getBooks().joinToString {
+      """
+      ${it.book.getTitle()}:
+      Articles: [${it.book.getArticleCount()}]
+      Creator: [${it.book.getCreator()}]
+      """.trimIndent()
+    }
+    return """
+        Current Zim File:
+        ${zimReaderContainer.zimCanonicalPath}
+        All Zim Files in DB:
+        $allZimFiles
+        
+        """.trimIndent()
+  }
+
+  private fun languageLocale(): String = """
+    Current Locale:
+    ${getCurrentLocale(applicationContext)}
+    
+  """.trimIndent()
+
+  private fun deviceDetails(): String = """
+    BluetoothClass.Device Details:
+    Device:[${Build.DEVICE}]
+    Model:[${Build.MODEL}]
+    Manufacturer:[${Build.MANUFACTURER}]
+    Time:[${Build.TIME}]
+    Android Version:[${Build.VERSION.RELEASE}]
+    App Version:[$versionName $versionCode]
+    
+  """.trimIndent()
+
+  private fun systemDetails(): String = """
+    Mount Points
+    ${mountPointProducer.produce().joinToString {
+    "$it\n"
+  }}
+    External Directories
+    ${externalFileDetails()}
+  """.trimIndent()
+
+  private fun externalFileDetails(): String = """
+    ${ContextCompat.getExternalFilesDirs(this, null).joinToString {
+    if (it != null)
+      it.path
+    else
+      "null"
+  }}
+  """.trimIndent()
+
+  private fun safeContains(extras: Bundle): Boolean {
+    return try {
+      extras.containsKey(EXCEPTION_KEY)
+    } catch (ignore: RuntimeException) {
+      false
+    }
+  }
+
+  protected open val subject: String
+    get() = "Someone has reported a crash"
+
+  protected open val initialBody: String
+    get() = """
+      Hi Kiwix Developers!
+      The Android app crashed, here are some details to help fix it:
+      """.trimIndent()
+
+  private val versionCode: Int
+    get() = packageManager
+      .getPackageInfo(packageName, ZERO).versionCode
+
+  private val versionName: String
+    get() = packageManager
+      .getPackageInfo(packageName, ZERO).versionName
+
+  private fun toStackTraceString(exception: Throwable): String = {
```

Comment:
this bracket pair should be unnecessary, should just read
```
private fun toStackTraceString(exception: Throwable): String =
```

Reply:
done